packages/components/src/gradient-picker/index.tsx

```diff
@@ -15,17 +15,21 @@ import CustomGradientPicker from '../custom-gradient-picker';
 import { VStack } from '../v-stack';
 import { ColorHeading } from '../color-palette/styles';
 import { Spacer } from '../spacer';
+import type {
+	GradientsProp,
+	GradientPickerComponentProps,
+	PickerProps,
+	OriginObject,
+	GradientObject,
+} from './types';
 
-// The Multiple Origin Gradients have a `gradients` property (an array of
-// gradient objects), while Single Origin ones have a `gradient` property.
-const isMultipleOriginObject = ( obj ) =>
-	Array.isArray( obj.gradients ) && ! ( 'gradient' in obj );
-
-const isMultipleOriginArray = ( arr ) => {
-	return (
-		arr.length > 0 &&
-		arr.every( ( gradientObj ) => isMultipleOriginObject( gradientObj ) )
-	);
+// For Single Origin Gradients, the `gradients` property
+// is an array of Gradient objects.
+// For Multiple Origin Gradients, the `gradients` property is an array of
+// objects, each with a `name` property and an internal `gradients` property,
+// which is the array of Gradient objects.
+const isMultipleOrigin = ( arr: GradientsProp ): arr is OriginObject[] => {
+	return ( arr as any[] ).every( ( obj ) => 'gradients' in obj );
```

Comment:
If we need the cast to `any[]` anyway, why don't we

```suggestion
const isMultipleOrigin = ( arr: any[] ): arr is OriginObject[] => {
	return arr.every( ( obj ) => 'gradients' in obj );
```

Reply:
That makes sense. At first I thought we'd want to keep the `GradientsProp` typing so we could ensure the function always received the right shape, even if we had to `as any[]` for the sake of getting `.every` through TS... but with this being an purely internal function that that isn't really a concern 👍